Core/GDCore/Events/Event.h

```diff
@@ -118,17 +118,17 @@ class GD_CORE_API BaseEvent {
   };
 
   /**
-   * Event must be able to return all comments std::vector they have.
-   * Used to preprocess the comments.
+   * Event must be able to return all strings std::vector they have.
+   * Used to preprocess the strings in an event.
    */
-  virtual std::vector<gd::EventsList*> GetAllCommentsVectors() {
-    std::vector<gd::EventsList*> noComments;
-    return noComments;
+  virtual std::vector<gd::String*> GetAllSearchableStrings() {
+    std::vector<gd::String*> noSearchableStrings;
+    return noSearchableStrings;
   };
-  virtual std::vector<const gd::EventsList*> GetAllCommentsVectors()
+  virtual std::vector<const gd::String*> GetAllSearchableStrings()
       const {
-    std::vector<const gd::EventsList*> noComments;
-    return noComments;
+    std::vector<const gd::String*> noSearchableStrings;
+    return noSearchableStrings;

```

Comment:
You have said "return empty string"
Then i return empty vector of string.
I guess all is ok here.
(Note for me Vector are like Array in JS)

Reply:
Yep vector is a resizeable array. "Vector" is basically an awful name for a "resizeable array" in C++ (because originally, the C arrays can't be resized).

You only need one method:
```
virtual std::vector<gd::String> GetAllSearchableStrings() const {
    std::vector<gd::String> noSearchableStrings;
    return noSearchableStrings;
}
```